pkg/controller/utils/discovery.go

```diff
@@ -167,15 +167,34 @@ func isDockerEE(ctx context.Context, c kubernetes.Interface) (bool, error) {
 // EKS doesn't have any provider-specific API groups, so we need to use a different approach than
 // we use for other platforms in autodetectFromGroup.
 func isEKS(ctx context.Context, c kubernetes.Interface) (bool, error) {
-	cm, err := c.CoreV1().ConfigMaps("kube-system").Get(ctx, "eks-certificates-controller", metav1.GetOptions{})
-	if err != nil {
-		if kerrors.IsNotFound(err) {
-			return false, nil
-		}
+	cm, err := c.CoreV1().ConfigMaps("kube-system").Get(ctx, "aws-auth", metav1.GetOptions{})
+	if err != nil && !kerrors.IsNotFound(err) {
 		return false, err
 	}
+	if cm != nil {
```

Comment:
I see you're just copying this logic from the existing code so that's fine but it looks odd to me because presence of this configmap is indicated by err being IsNotFound. I don't think cm being nil is behavior that's meant to be relied on for anything.

Fine to merge as is or take a crack at refactoring, either is ok to me